### PR TITLE
[memprof] Add flag to control profile dump at exit

### PR DESCRIPTION
**Pull request by ellishg**
Add the `dump_at_exit` flag to control whether or not profiles should be dumped when the program exits. Since we can call `__memprof_profile_dump()` directly, we don't necessarily need to dump profiles at exit.

### CODE REVIEW

File: compiler-rt/test/memprof/TestCases/dump_at_exit.cpp [hunk 1]
```diff
@@ -0,0 +1,16 @@
+// RUN: %clangxx_memprof %s -o %t
+
+// RUN: %env_memprof_opts=print_text=true:log_path=stdout:dump_at_exit=false %run %t | count 0
```
Comment (snehasish) [inline compiler-rt/test/memprof/TestCases/dump_at_exit.cpp:3]: What is `count 0` here?

**Review by snehasish — COMMENTED** [review]
Can you describe the scenario where this is useful?

File: compiler-rt/test/memprof/TestCases/dump_at_exit.cpp (hunk 1 above)
Comment (ellishg) [reply compiler-rt/test/memprof/TestCases/dump_at_exit.cpp:3]: `count N` will pass if it reads `N` lines from stdin. So this just checks that there is no output.

Comment (snehasish) [reply compiler-rt/test/memprof/TestCases/dump_at_exit.cpp:3]: Ah count is a utility we bundle with LLVM for testing. TIL.

**Review by snehasish — APPROVED** [review]
lgtm

**Pull request merged**